src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -169,308 +170,34 @@ static int mixin_free(struct processing_module *mod)
 
 	comp_dbg(mod->dev, "mixout_free()");
-
-	mod_source_info = module_source_info_acquire(mod->source_info);
-	rfree(mod_source_info->private);
-	mod_source_info->private = NULL;
-	module_source_info_release(mod_source_info);
+	rfree(md);
 
 	return 0;
 }
 
-static int mix_and_remap(struct comp_dev *dev, const struct mixin_data *mixin_data,
-			 uint16_t sink_index, struct audio_stream __sparse_cache *sink,
-			 uint32_t start_frame, uint32_t mixed_frames,
-			 const struct audio_stream __sparse_cache *source, uint32_t frame_count)
+static int mixout_process_single_source(struct processing_module *mod,
+					struct input_stream_buffer *input_buffer,
+					struct output_stream_buffer *output_buffer)
 {
-	const struct mixin_sink_config *sink_config;
 
-	if (sink_index >= MIXIN_MAX_SINKS) {
-		comp_err(dev, "Sink index out of range: %u, max sinks count: %u",
-			 (uint32_t)sink_index, MIXIN_MAX_SINKS);
-		return -EINVAL;
-	}
+	audio_stream_copy(input_buffer->data, 0, output_buffer->data, 0,
+			  input_buffer->size * audio_stream_get_channels(input_buffer->data));
 
-	sink_config = &mixin_data->sink_config[sink_index];
-
-	if (sink_config->mixer_mode == IPC4_MIXER_NORMAL_MODE) {
-		/* Mix streams. mix_channel() is reused here to mix streams, not individual
-		 * channels. To do so, (multichannel) stream is treated as single channel:
-		 * channel count is passed as 1, channel index is 0, frame indices (start_frame
-		 * and mixed_frame) and frame count are multiplied by real stream channel count.
-		 */
-		mixin_data->normal_mix_channel(sink, start_frame * audio_stream_get_channels(sink),
-					       mixed_frames * audio_stream_get_channels(sink),
-					       source,
-					       frame_count * audio_stream_get_channels(sink),
-					       sink_config->gain);
-	} else if (sink_config->mixer_mode == IPC4_MIXER_CHANNEL_REMAPPING_MODE) {
-		int i;
-
-		for (i = 0; i < audio_stream_get_channels(sink); i++) {
-			uint8_t source_channel =
-				(sink_config->output_channel_map >> (i * 4)) & 0xf;
-
-			if (source_channel == 0xf) {
-				mixin_data->mute_channel(sink, i, start_frame, mixed_frames,
-							 frame_count);
-			} else {
-				if (source_channel >= audio_stream_get_channels(source)) {
-					comp_err(dev, "Out of range chmap: 0x%x, src channels: %u",
-						 sink_config->output_channel_map,
-						 audio_stream_get_channels(source));
-					return -EINVAL;
-				}
-				mixin_data->remap_mix_channel(sink, i,
-							      audio_stream_get_channels(sink),
-							      start_frame, mixed_frames,
-							      source, source_channel,
-							      audio_stream_get_channels(source),
-							      frame_count, sink_config->gain);
-			}
-		}
-	} else {
-		comp_err(dev, "Unexpected mixer mode: %d", sink_config->mixer_mode);
-		return -EINVAL;
-	}
+	input_buffer->consumed =
+		input_buffer->size * audio_stream_frame_bytes(input_buffer->data);
+	output_buffer->size =
+		input_buffer->size * audio_stream_frame_bytes(output_buffer->data);
```

Comment:
since this is a simple copy, so the frame size of input & output is the same, so how about 
frame_size = audio_stream_frame_bytes(input_buffer->data);
	input_buffer->consumed = input_buffer->size * frame_size;
	output_buffer->size = input_buffer->size * frame_size;